### PR TITLE
Refacto contact page

### PR DESCRIPTION
**Pull request by LuckyShuii**

### CODE REVIEW

**Comment by LuckyShuii** [comment]
I messed up again Jesus .. ignore the articles files in this PR they are already present in an other PR

File: public/assets/js/contact.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+let btn_envoie = document.querySelector("#btn_submit")
```
Comment (Richarddesnoues) [inline public/assets/js/contact.js:1]: variable en français ?

File: public/assets/js/contact.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+let inputs = document.querySelectorAll(".input_form")
+let label_mod = document.querySelector(".labelMod")
+let contact_form = document.querySelector(".contactUsForm")
+let check_form = []
+
+btn_envoie.addEventListener("click", (event) => {
```
Comment (Richarddesnoues) [inline public/assets/js/contact.js:7]: là aussi

File: public/assets/js/contact.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            case "text":
+                if (elementID === "prenom" || elementID === "nom" || elementID === "subject") {
+                    if (valueInput === "") {
+                        error_input(element);
+                    } else {
+                        element.style.borderBottom = "1px solid black";
+                        element.nextElementSibling.style.display = "none";
+                    }
+
+                    if (elementID === "prenom" && valueInput.length < 2) {
```
Comment (Richarddesnoues) [inline public/assets/js/contact.js:32]: FRANCAIS ! !

File: public/assets/js/contact.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                        error_input(element);
+                    }
+
+                    if (elementID === "nom" && valueInput.length < 2) {
```
Comment (Richarddesnoues) [inline public/assets/js/contact.js:36]: FRANCAIS !!!

**Review by Richarddesnoues — CHANGES_REQUESTED** [review]

File: public/assets/js/contact.js (hunk 2 above)
Comment (LuckyShuii) [reply public/assets/js/contact.js:7]: changed 👉🏽 next commit

File: public/assets/js/contact.js (hunk 1 above)
Comment (LuckyShuii) [reply public/assets/js/contact.js:1]: changed 👉🏽 next commit

File: public/assets/js/contact.js (hunk 3 above)
Comment (LuckyShuii) [reply public/assets/js/contact.js:32]: changed 👉🏽 next commit

File: public/assets/js/contact.js (hunk 4 above)
Comment (LuckyShuii) [reply public/assets/js/contact.js:36]: changed 👉🏽 next commit

**Review by Richarddesnoues — APPROVED** [review]

File: src/Controller/ArticleController.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+class ArticleController extends AbstractController
+{
+    /**
+     * Display home page
+     */
+    public function show(int $id): string
+    {
+        $articleManager = new ArticleManager();
+        $article = $articleManager->selectOneById($id);
```
Comment (Benja-san) [inline src/Controller/ArticleController.php:16]: getArticle now ?

File: src/Controller/ContactController.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            $errors
+        );
+        $this->validateStringLength('lastName', 2, PHP_INT_MAX, 'Le nom doit contenir au moins 2 caractères', $errors);
+        $this->validateStringLength('topic', 4, 80, 'Le sujet doit contenir entre 4 et 80 caractères', $errors);
+        $this->validateCheckbox('validation', 'Vous devez accepter les conditions d\'utilisation', $errors);
+
+        return $errors;
+    }
+
+    private function validateRequiredFields(array $fields, array &$errors): void
```
Comment (Benja-san) [inline src/Controller/ContactController.php:56]: is the & before error relevant ?

**Review by Benja-san — APPROVED** [review]

**Pull request merged**